lib/galaxy/datatypes/binary.py

```diff
@@ -736,6 +736,97 @@ def display_peek(self, dataset):
+    MetadataElement(name="row_attrs_count", default=0, desc="row_attrs_count", readonly=True, visible=True, no_value=0)
+    MetadataElement(name="row_attrs_names", desc="row_attrs_names", default=[], param=metadata.SelectParameter, multiple=True, readonly=True, no_value=None)
+    MetadataElement(name="col_attrs_count", default=0, desc="col_attrs_count", readonly=True, visible=True, no_value=0)
+    MetadataElement(name="col_attrs_names", desc="col_attrs_names", default=[], param=metadata.SelectParameter, multiple=True, readonly=True, no_value=None)
+    MetadataElement(name="col_graphs_count", default=0, desc="col_graphs_count", readonly=True, visible=True, no_value=0)
+    MetadataElement(name="col_graphs_names", desc="col_graphs_names", default=[], param=metadata.SelectParameter, multiple=True, readonly=True, no_value=None)
+    MetadataElement(name="row_graphs_count", default=0, desc="row_graphs_count", readonly=True, visible=True, no_value=0)
+    MetadataElement(name="row_graphs_names", desc="row_graphs_names", default=[], param=metadata.SelectParameter, multiple=True, readonly=True, no_value=None)
+
+    def sniff(self, filename):
+        if super(Loom, self).sniff(filename):
+            try:
+                with h5py.File(filename) as loom_file:
+                    return bool(loom_file.attrs.get('LOOM_SPEC_VERSION', False))
+            except Exception:
+                return False
+        return False
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Binary Loom file"
+            dataset.blurb = nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Binary Loom file (%s)" % (nice_size(dataset.get_size()))
+
+    def set_meta(self, dataset, overwrite=True, **kwd):
+        super(Loom, self).set_meta(dataset, overwrite=overwrite, **kwd)
+        try:
+            with h5py.File(dataset.file_name) as loom_file:
+                dataset.metadata.title = loom_file.attrs.get('title', None)
+                dataset.metadata.description = loom_file.attrs.get('description', None)
+                dataset.metadata.url = loom_file.attrs.get('url', None)
+                dataset.metadata.doi = loom_file.attrs.get('doi', None)
+                dataset.metadata.loom_spec_version = loom_file.attrs.get('LOOM_SPEC_VERSION', None)
+                dataset.creation_date = loom_file.attrs.get('creation_date', None)
+                dataset.metadata.shape = tuple(loom_file['matrix'].shape)
+
+                tmp = list(loom_file['layers'].keys())
+                dataset.metadata.layers_count = len(tmp)
+                dataset.metadata.layers_names = tmp
```

Comment:
If loom_file['layers'] isn't osme kind of OrderedDict we should probably sort here

Reply:
Its not technically a dictionary at all. Its an object that implements several bits of the dictionary style interface. My understanding is that h5py returns things in the same order as the standard tools (alphanumerically sorted), which is probably the desired behavior here, but I haven't done any extensive testing here. Presumably, if a specific tool needs a different ordering, then a sort filter could be applied within the parameter dropdown. Although if we really think we need to manually resort things, we can.

Reply:
I'm thinking about tool tests and equality of metadata when caching jobs here. If some of the metadata elements are returned with a random order that might be a little annoying. If it's the same order it's fine for me, and we can always fix it later.